metrotranscom/doorway
# fix: add db:reseed:ci

# Pull Request Template

## Issue Overview


## Description

There's a chicken-and-egg situation that happens with reseeding.  The first thing it does is try to drop the `bloom` table, but if that's the table you're connected to when you run the command, it fails.  This is less an issue in local dev environments where there are alternate databases to use, but in CI environments that isn't necessarily an option and automating around it can be a pain.  This PR adds a new yarn command, `db:reseed:ci`, that addresses this issue by automatically creating a new temporary database and running database reset commands from there before reseeding.

Documentation for this and other commands will be added later, once more important tasks are complete.

## How Can This Be Tested/Reviewed?

Run `yarn db:reseed:ci` and verify that it completes successfully.  You can also verify that the `temp` database is not used for populating seed data by running `psql -d temp -c '\d'`, which should print "Did not find any relations."  If your `bloom` database doesn't yet exist for some reason, you can change the starting database by setting `PGDATABASE` ahead of the command like this: `PGDATABASE=<other_db_name> yarn db:reseed:ci`.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
